core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java

```diff
@@ -0,0 +1,42 @@
+package com.aembootcamp.core.models;
+
+import com.adobe.cq.wcm.core.components.models.Breadcrumb;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.Via;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.apache.sling.models.annotations.injectorspecific.SlingObject;
+import org.apache.sling.models.annotations.via.ResourceSuperType;
+import lombok.experimental.Delegate;
+
+@Model(adaptables = SlingHttpServletRequest.class, adapters = Breadcrumb.class, resourceType = BreadcrumbModel.RESOURCE_TYPE)
+public class BreadcrumbModel implements Breadcrumb {
+
+    protected static final String RESOURCE_TYPE = "aem-bootcamp/components/structure/breadcrumb";
+
+    @SlingObject
+    private Resource currentResource;
```

Comment:
Validate if you can change the currentResource with:

@SlingObject
private Page currentPage;